src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -1200,9 +1207,18 @@ private void parseAttachments(final Element root) throws GameParseException {
               () -> new GameParseException(mapName, "Attachment of type " + className + " could not be instantiated"));
       attachable.addAttachment(name, attachment);
 
-      final ArrayList<Tuple<String, String>> attachmentOptionValues = setValues(attachment, options);
       // keep a list of attachment references in the order they were added
-      data.addToAttachmentOrderAndValues(Tuple.of(attachment, attachmentOptionValues));
+      executor.submit(() -> Tuple.of(attachment, setValues(attachment, options)));
```

Comment:
Let's take a step back for a moment...  What exactly are we trying to speed up here via parallelization?  Is it specifically the XML parsing that's slow or applying the parsed values to the attachment?

If the former, it might be easier to have each executor task parse the XML and build some kind of immutable data structure that contains the parsed values, which can then be applied relatively quickly to the attachment.  Each task would pass that immutable data structure back to the main thread, where the attachments would be modified as data becomes available.  Since the attachments will be modified on a single thread, that should obviate the need to add all the ad-hoc synchronization present in this PR.

Ideally, all of these parallel tasks should not operate on any shared data, if possible.  They perform some work and return it to the main thread, which is then responsible for composing all the individual results into the final result.  If we can do something like that, it will be much easier to reason about and get correct.

<hr>

Note also that running `setValues()` in a different thread means the DOM will be accessed concurrently.  IIRC, the default JDK DOM implementation is Xerces, which is [not thread-safe](https://xerces.apache.org/xerces2-j/faq-dom.html#faq-1).

Reply:
The dom is already completely parsed at this point, we are parallelizing the application of attachment values.
This is possible, because the different attachment types normally don't affect each other, so we can execute them asynchronously.
Thread-Safety shouldn't be that important, because we only do read operations at this point.